11-trt-llm/README.md

```diff
@@ -0,0 +1,75 @@
+- `max_input_len`: Maximum input sequence length
+- `max_output_len`: Maximum output sequence length
+- `max_batch_size`: Maximum batch size for inference
+- `max_beam_width`: Maximum beam width for beam search
+- `huggingface_ckpt_repository`: Path to the HuggingFace model checkpoint with the PyTorch weights
+
+Other optional fields control quantization, parallelism, and plugin settings.
+
+The `serve` section specifies the paths to the compiled TRT engine and tokenizer to use for serving the model. 
+
+Note that build and serve are mutually exclusive - you can only specify one or the other in the config.
+
+### Quantization
+One of the features we support is quantization. Quantization is a powerful way to improve latency, TTFT and throughput without significantly dropping the quality performance of the model. We offer the following forms of quantization:
+
+The `quantization_type` field in the `trt_llm` config specifies the quantization method to be used when building the TRT engine:
+
+- `NO_QUANT`: No quantization is applied, and the model runs with full precision (FP32).
+- `WEIGHTS_ONLY_INT8`: Only the model weights are quantized to INT8 precision, while activations remain in FP32.
+- `WEIGHTS_KV_INT8`: Both the model weights and the key-value cache (used for attention mechanisms) are quantized to INT8.
+- `WEIGHTS_ONLY_INT4`: Model weights are quantized to INT4 precision.
+- `WEIGHTS_KV_INT4`: Both weights and key-value cache are quantized to INT4.
+- `SMOOTH_QUANT`: A quantization technique that smooths the quantized weights to improve accuracy.
+- `FP8`: Quantization to FP8 precision (8-bit floating point).
+- `FP8_KV`: Both weights and key-value cache are quantized to FP8.
+
+In general, lower precision quantization (e.g., INT4) results in smaller model size and faster inference, but may suffer from more significant accuracy degradation. Higher precision quantization (e.g., INT8, FP8) strikes a better balance between performance and accuracy.
+
+### Serving Architecture
+![image](./architecture.png)
+
+Triton Inference Server operates with protobuf-like inputs. This is highly performant but can result in a difficult user experience. This Truss uses the `TrussServer` to proxy JSON requests over a GRPC stream to the Triton Infernece Server. 
+
+The model repository for Triton is located under the `packages` directory: `packages/tensorrt_llm_model_repository`, In classic TRT-LLM fashion, we run the model as an ensemble of models:
+- A `preprocess` model that tokenizes inputs
+- A `tensorrt_llm` model that uses the TensorRT-LLM backend for Triton to run your engine. During model startup, we move the engine into `/packages/tensorrt_llm_model_repository/tensorrt_llm/1/`. 
+- A `postprocess` model that handles streaming detokenization of tokens produced by the previous model. 
+
+These tokens are streamed to the Truss Server (defined in `model/model.py`) and then streamed to the user. 
```

Comment:
After reading this doc, it's not clear what action I'm supposed to take. What is the end goal of a reader on this doc and what steps do they need to take to get there?